### PR TITLE
[BUG FIX] fix cmake/pc files: enable linking from outside.

### PR DESCRIPTION
**Pull request by fghoussen**
## Pull request purpose

fixing issue #214 (at least partially).

### CODE REVIEW

File: arpack.pc.in [hunk 1]
```diff
@@ -7,6 +7,5 @@ Name: @PACKAGE_NAME@
 Description: Collection of Fortran77 subroutines designed to solve large scale eigenvalue problems
 Version: @PACKAGE_VERSION@
 URL: @PACKAGE_URL@
-Libs: -L${libdir} -larpack@LIBSUFFIX@
-Libs.private: @LAPACK_LIBS@ @BLAS_LIBS@
-Cflags: -I${includedir}/arpack
+Libs: -L${libdir} -larpack@LIBSUFFIX@ @LAPACK_LIBS@ @BLAS_LIBS@
```
Comment (QuLogic) [inline arpack.pc.in]: This is wrong; LAPACK and BLAS are private. Moving them to public causes overlinking of libraries that use arpack. If you are using a static build of arpack, you need to use `pkg-config --static` to grab the private flags.

Comment (sylvestre) [reply arpack.pc.in]: what do you mean by private ?

Comment (QuLogic) [reply arpack.pc.in]: Private meaning the `Libs.private` field that they used to be in.

Comment (fghoussen) [reply arpack.pc.in]: Couldn't link without it at the time. Adding non reg test for that would be good.

Comment (sylvestre) [reply arpack.pc.in]: ok, please provide patches :)

**Pull request merged**